resources/deployments/mongo/device_deployments.go

```diff
@@ -433,7 +433,7 @@ func (d *DeviceDeploymentsStorage) DecommissionDeviceDeployments(deviceId string
 	_, err := session.DB(DatabaseName).C(CollectionDevices).UpdateAll(selector, update)
```

Comment:
also did a POC:

if the selector doesn't match it's not an error - regardless of the SafeMode.